Source/Particles/Collision/MCCScattering.H

```diff
@@ -0,0 +1,304 @@
+    * should undergo an ionization collision.
+    *
+    * @param[in] ptd particle tile data
+    * @param[in] i particle index
+    * @return true if a collision occurs, false otherwise
+    */
+    template <typename PData>
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    bool operator() (
+        const PData& ptd, int const i, amrex::RandomEngine const& engine
+    ) const noexcept
+    {
+        using namespace amrex;
+        using std::sqrt;
+
+        // determine if this particle should collide
+        if (Random(engine) > m_total_collision_prob) return false;
+
+        // get the particle velocity
+        const ParticleReal ux = ptd.m_rdata[PIdx::ux][i];
+        const ParticleReal uy = ptd.m_rdata[PIdx::uy][i];
+        const ParticleReal uz = ptd.m_rdata[PIdx::uz][i];
+
+        // calculate kinetic energy
+        const ParticleReal v_coll2 = ux*ux + uy*uy + uz*uz;
+        const ParticleReal E_coll = 0.5 * m_mass * v_coll2 / PhysConst::q_e;
+        const ParticleReal v_coll = sqrt(v_coll2);
+
+        // get collision cross-section
+        const Real sigma_E = m_mcc_process.getCrossSection(E_coll);
+
+        // calculate normalized collision frequency
+        const Real nu_i = sigma_E * v_coll / m_nu_max_norm;
+
+        // check if this collision should be performed
+        return (Random(engine) <= nu_i);
+    }
+
+private:
+    MCCProcess m_mcc_process;
+    amrex::Real m_mass;
+    amrex::Real m_total_collision_prob = 0;
+    amrex::Real m_nu_max_norm;
+};
+
+
+/**
+ * \brief Transform functor for impact ionization
+ */
+class ImpactIonizationTransformFunc
+{
+public:
+
+    /**
+    * \brief Constructor of the ImpactIonizationTransformFunc functor.
+    *
+    * The transform is responsible for appropriately decreasing the kinetic
+    * energy of the colliding particle and assigning appropriate velocities
+    * to the two newly created particles. To this end the energy cost of
+    * ionization is passed to the constructor as well as the mass of the
+    * colliding species and the standard deviation of the ion velocity
+    * (normalized temperature).
+    *
+    * @param[in] energy_cost energy cost of ionization
+    * @param[in] mass1 mass of the colliding species
+    */
+    ImpactIonizationTransformFunc(
+        amrex::Real energy_cost, amrex::Real mass1, amrex::Real ion_vel_std
+    ) :  m_energy_cost(energy_cost), m_mass1(mass1),
+         m_ion_vel_std(ion_vel_std) { }
+
+    /**
+    * \brief Functor call. It determines the properties of the generated pair
+    * and decreases the kinetic energy of the colliding particle. Inputs are
+    * standard from the FilterCopyTransfrom::filterCopyTransformParticles
+    * function.
+    *
+    * @param[in,out] dst1 target species 1 (either electrons or positrons)
+    * @param[in,out] dst2 target species 2 (either electrons or positrons)
+    * @param[in] src source species (photons)
```

Comment:
Did I understand correctly that the expected species are the ones below
```suggestion
    * @param[in,out] dst1 target species 1 (electrons)
    * @param[in,out] dst2 target species 2 (ions)
    * @param[in] src source species (electrons)
```

Reply:
Yeap, this is just a oversight from copy-paste. Thanks for catching it!